File: app/Models/Room.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Room extends Model
{
    protected $fillable = [
        'property_id',
        'room_number',
        'floor',
        'status',
    ];

    public function property()
    {
        return $this->belongsTo(Property::class);
    }

    public function bookings()
    {
        return $this->hasMany(Booking::class, 'assigned_room_id');
    }
}

File: frontend/lib/models/room.dart
class Room {
  final int? id;
  final int propertyId;
  final String roomNumber;
  final String? floor;
  final String status;

  Room({
    this.id,
    required this.propertyId,
    required this.roomNumber,
    this.floor,
    required this.status,
  });

  factory Room.fromJson(Map<String, dynamic> json) {
    return Room(
      id: json['id'],
      propertyId: json['property_id'],
      roomNumber: json['room_number'],
      floor: json['floor'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'property_id': propertyId,
      'room_number': roomNumber,
      'floor': floor,
      'status': status,
    };
  }
}
